### PR TITLE
Fix bug setting the base sha for the new branch

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@brettfo @dotnet/roslyn-infrastructure

File: src/Tools/Github/GitMergeBot/Program.cs [hunk 1]
```diff
@@ -158,13 +158,14 @@ private Program(Options options)
...
 ``` bash
 git remote add {remoteName} ""https://github.com/{_options.SourceUser}/{_options.RepoName}.git""
+# git remote add {remoteName} ""git@github.com:{_options.SourceUser}/{_options.RepoName}.git""
 git fetch {remoteName}
-git checkout {newBranchName}
-git reset --hard upstream/{_options.DestinationBranch}
-git merge upstream/{_options.SourceBranch}
+git checkout {newBranchName} # Switch to the pr-branch
+git pull upstream/{_options.DestinationBranch} --ff-only # Get the most recent commits
+git merge upstream/{_options.SourceBranch}
```
Comment (brettfo) [inline src/Tools/Github/GitMergeBot/Program.cs]: Extra whitespace on the end of the line?

**Comment by brettfo** [comment]
:shipit: pending verification of the PRs it creates.

**Comment by davkean** [comment]
(Closing/opening to sync update the merge branch to pick up the build break we just fixed)

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/dbg/unit64 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/2691/
// Retest reason: build break

**Comment by TyOverby** [comment]
Closing because it turns out that github doesn't support this feature

**Pull request closed without merging**